core/src/test/java/org/testcontainers/junit/OutputStreamWithTTYTest.java

```diff
@@ -0,0 +1,94 @@
+package org.testcontainers.junit;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.output.ToStringConsumer;
+import org.testcontainers.containers.output.WaitingConsumer;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.*;
+import static org.testcontainers.containers.output.OutputFrame.OutputType.STDOUT;
+
+public class OutputStreamWithTTYTest {
+
+    @Rule
+    public GenericContainer container = new GenericContainer("alpine:3.2")
+        .withCommand("/bin/sh", "-c", "sleep 2; ls -1" +
+            "")
+        .withCreateContainerCmdModifier(command -> ((CreateContainerCmd)command).withTty(Boolean.TRUE));
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OutputStreamTest.class);
+
+    @Test
+    public void testFetchStdout() throws TimeoutException {
+
+        WaitingConsumer consumer = new WaitingConsumer();
+
+        container.followOutput(consumer, STDOUT);
+
+        consumer.waitUntil(frame -> frame.getType() == STDOUT && frame.getUtf8String().contains("home"),
+            30, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testFetchStdoutWithTimeout() throws TimeoutException {
+
+        WaitingConsumer consumer = new WaitingConsumer();
+
+        container.followOutput(consumer, STDOUT);
+
+        assertThrows("a TimeoutException should be thrown", TimeoutException.class, () -> {
+            consumer.waitUntil(frame -> frame.getType() == STDOUT && frame.getUtf8String().contains("qqq"),
+                4, TimeUnit.SECONDS);
```

Comment:
is it necessary to be 4 seconds?

Reply:
It's can be 3 seconds :smiley: Just should be greater than sleep in running script.